src/pymatgen/electronic_structure/dos.py

```diff
@@ -822,7 +822,7 @@ def get_band_filling(
 
         # Only consider up to Fermi level in numerator
         energies = dos.energies - dos.efermi
-        return np.trapz(dos_densities[energies < 0], x=energies[energies < 0]) / np.trapz(
+        return np.trapezoid(dos_densities[energies < 0], x=energies[energies < 0]) / np.trapezoid(
```

Comment:
pre-commit seems to update the `np.trapz` with new `np.trapezoid` in NumPy 2, and would break CI tests